modules/core/src/main/scala/org/scalasteward/core/application/Context.scala

```diff
@@ -193,6 +194,9 @@ object Context {
         .create[F, Repo, RepoCache]("repo_cache", "1", kvsPrefix)
       versionsStore <- JsonKeyValueStore
         .create[F, VersionsCache.Key, VersionsCache.Value]("versions", "2")
+      gitlabUserIdCacheStore <- JsonKeyValueStore
+        .create[F, String, Int]("gitlab_user_id_cache", "1", kvsPrefix)
+        .flatMap(CachingKeyValueStore.wrap(_))
```

Comment:
**question:** Could you please motivate why you added a cache for the user name to ID mapping. I understand that it avoids repeated API calls during PR creation but is a cache for that really necessary? Is it expensive or slow to get the IDs by a username? How much slower would Scala Steward be if these API calls are repeatedly done per PR creation?

I'm asking because adding this cache is not for free and from my perspective this optimization also looks like a complication and I don't know why it is necessary.

Reply:
When I was playing around this API I have not noticed any delays or slow responses - so, I would say these calls to `GET /users` are lightweight. I've added cache because `username -> user ID` mapping is something that does not change often if changed ever at all, rather than `GET /users/ being expensive in terms of time or something else.

I've checked again now and for my location calls to `GET /users` on public gitlab.com API takes ~400ms on average, same as for `GET /merge_requests`.

I run Scala Steward once in two weeks typically and it creates usually around 20 pull requests. For example, last week it have created 22 PRs and it took approximately 5 minutes (only API interaction stage). If I will add 5 assignees to the config then Steward will issue 5 extra calls for each PR (for example 400ms each call, giving 2 extra seconds for each PR) that will give extra 40 seconds for the whole run.

I am not sure it's enough to actually decide to keep the cache since - (1) it's not dramatically big slowdown for the described workflow, (2) the code gets more complex with the cache, (3) if the feature will introduce critical slow down for some workflows it's always possible to do optimization in the future.

On the other hand, I would say that caching here is not extraordinary complex or tricky or decreases readability dramatically while can bring noticeable optimization. 

So, what do you think at that point, should we keep or remove cache finally? My feelings are that I do not mind to either remove or keep it - for me both options are pretty equal

P.S. another option will be to introduce some in-memory caching to avoid storing the data to the disk, but I've not found any adopted solutions for such caching in the codebase so it will require to bring something new. The reasonable thing to do maybe will be to reuse `KeyValueStore` interface and create in-memory implementation for it.

Reply:
Also take into account the rate limits the GitHub API has, this cache will ensure we won't go over the roof with those

Reply:
😓 That's right! I totally missed that. Thanks for the clarification!

Reply:
Cache dropped :+1: